### PR TITLE
Added a preview page to view all image styles

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Preview page of all image styles.

# Needed By (Date)
- whenever

# Urgency
- not at all

# Steps to Test

1. Checkout branch.
2. enable `stanford_image_styles_preview` module
3. navigate to the image styles page.
4. Click on the "Preview Styles" tab at the top.
5. view the default image styles
6. upload your own image  and view the styles.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: modules/stanford_image_styles_preview/src/Form/PreviewForm.php [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+/**
+ * Class PreviewForm.
+ */
+class PreviewForm extends FormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'preview_form';
```
Comment (sherakama) [inline modules/stanford_image_styles_preview/src/Form/PreviewForm.php]: The class is namespaced well but should we prepend the module name to this id?

File: modules/stanford_image_styles_preview/src/Form/PreviewForm.php [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $form = [];
```
Comment (sherakama) [inline modules/stanford_image_styles_preview/src/Form/PreviewForm.php]: $form is already defined. Why reset it to an empty array?

File: modules/stanford_image_styles_preview/src/Form/PreviewForm.php [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+    $form['image'] = [
+      '#type' => 'managed_file',
+      '#title' => $this->t('Image'),
+      '#description' => $this->t('Upload an image to see how the image styles react'),
+      '#required' => TRUE,
+      '#upload_validators' => [
```
Comment (sherakama) [inline modules/stanford_image_styles_preview/src/Form/PreviewForm.php]: This is cool

File: modules/stanford_image_styles_preview/src/Form/PreviewForm.php [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+   * Builds the preview portions of the form using a default image or an
+   * uploaded image.
+   *
+   * @param array $form
+   *   Simple form api array.
+   * @param FormStateInterface $form_state
+   *   Form State from building the form.
+   */
+  private function buildPreview(array &$form, FormStateInterface &$form_state) {
+    $style_ids = \Drupal::entityQuery('image_style')->execute();
```
Comment (sherakama) [inline modules/stanford_image_styles_preview/src/Form/PreviewForm.php]: I'm surprised CC isn't throwing the dependency injection error here.

File: modules/stanford_image_styles_preview/src/Form/PreviewForm.php [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $form = [];
+
+    $form['image'] = [
+      '#type' => 'managed_file',
```
Comment (sherakama) [inline modules/stanford_image_styles_preview/src/Form/PreviewForm.php]: Do we wan't a managed file here? Seems that the form is only temporarily using the uploaded image. Managed files stick around.

File: modules/stanford_image_styles_preview/src/Form/PreviewForm.php (hunk 2 above)
Comment (pookmish) [inline modules/stanford_image_styles_preview/src/Form/PreviewForm.php]: just force of habbit i suppose

File: modules/stanford_image_styles_preview/src/Form/PreviewForm.php (hunk 4 above)
Comment (pookmish) [inline modules/stanford_image_styles_preview/src/Form/PreviewForm.php]: I dont think we have CC turned on for this repo

File: modules/stanford_image_styles_preview/src/Form/PreviewForm.php (hunk 5 above)
Comment (pookmish) [inline modules/stanford_image_styles_preview/src/Form/PreviewForm.php]: I tried a normal file but with a required field it wouldn't validate. I wasn't sure why. I can try to see if theres a work around

Comment (sherakama) [inline modules/stanford_image_styles_preview/src/Form/PreviewForm.php]: Why not remove the required portion then? Add a validation method that sets a message for the end user if the image is missing instead.

Comment (pookmish) [inline modules/stanford_image_styles_preview/src/Form/PreviewForm.php]: the wierd thing is that during validation or submit, i'm not seeing any info about the file location or anything. not sure what im missing.

File: modules/stanford_image_styles_preview/src/Form/PreviewForm.php (hunk 4 above)
Comment (sherakama) [inline modules/stanford_image_styles_preview/src/Form/PreviewForm.php]: Ah, that'd be why.

File: modules/stanford_image_styles_preview/src/Form/PreviewForm.php (hunk 5 above)
Comment (pookmish) [inline modules/stanford_image_styles_preview/src/Form/PreviewForm.php]: So during validation, I can save the file but it becomes a managed file that way as well. I'm unable to find the path to the file even if it was a temporary path. 
If we keep it as a file_managed field, the uploaded file _should_ be cleaned up during garbage cleanup.
Another option would be to eliminate the upload and just have a few stock images included in the module and allow switching the preview.

Comment (sherakama) [inline modules/stanford_image_styles_preview/src/Form/PreviewForm.php]: My concern is more with orphaned files floating around the file system/table. If you can confirm that the image does get garbage collected I am good with this.

**Comment by sherakama** [comment]
@pookmish does this clean up the uploaded images? If so I am happy to merge.

**Comment by pookmish** [comment]
@sherakama i just confirmed that the file gets cleaned up during cron. I had to edit the `created` and `updated` fields in the database to simulate an old file but then it removed on the next cron run.

**Comment by sherakama** [comment]
:1st_place_medal:

**Pull request merged**